pkg/events/watcher.go

```diff
@@ -106,7 +108,7 @@ func (w *eventWatcher) Reconcile(ctx context.Context, request reconcile.Request)
 			return reconcile.Result{}, err
 		}
 	default:
-		panic("invalid event")
+		return reconcile.Result{}, eris.Errorf("invalid event")
```

Comment:
Do you think it makes sense to include any context around the event type so that if it occurs we can more easily debug it?